Cześć Maciek,

deseff opened the pull request:
nie wychodzi mi test, gdy komentarzy jest mniej niż postów. Czy mógłbyś zerknąć i coś podpowiedzieć?

Module 6.5 - Zadanie: mockujemy forum dyskusyjne
             Wyobraź sobie, że masz za zadanie dopisanie modułu prezentującego statystyki forum dyskusyjnego. Forum jeszcze nie jest ukończone, masz jednak specyfikację interfejsu, przy pomocy którego będziesz pobierać dane. Oto on (utwórz sobie pakiet com.kodilla.testing.forum.statistics w katalogu src/main/java):

             package com.kodilla.testing.forum.statistics;

             import java.util.List;

             public interface Statistics {
                 List<String> usersNames(); // list of users names
                 int postsCount();          // total quantity of forum posts
                 int commentsCount();       // total quantity of forum comments
             }
             Pamiętaj, że brak public przed nazwą metody w interfejsie również oznacza, że jest ona publiczna. Interfejs nie może posiadać innych metod, niż publiczne, więc nie trzeba tego pisać.

             W pakiecie com.kodilla.testing.forum.statistics utwórz klasę obliczającą następujące statystyki:

             liczbę użytkowników,
             liczbę postów,
             liczbę komentarzy,
             średnią liczbę postów na użytkownika,
             średnią liczbę komentarzy na użytkownika,
             średnią liczbę komentarzy na post.
             Stworzona klasa powinna posiadać metodę calculateAdvStatistics(Statistics statistics), która obliczy podane powyżej wartości i zapamięta je we właściwościach (polach) klasy. Możesz również dopisać metodę showStatistics(), która wyświetli zapamiętane we właściwościach statystyki.

             Następnie utwórz pakiet com.kodilla.testing.forum.statistics w katalogu src/test/java i stwórz zestaw testów, zawierający testy sprawdzające metodę calculateAdvStatistics(Statistics statistics) testowanej klasy. Testy zrealizuj przy pomocy mocka interfejsu Statistics. Przetestuj poprawność obliczeń wartości średnich dla różnych przypadków:

             gdy liczba postów = 0,
             gdy liczba postów = 1000,
             gdy liczba komentarzy = 0,
             gdy liczba komentarzy < liczba postów,
             gdy liczba komentarzy > liczba postów,
             gdy liczba użytkowników = 0,
             gdy liczba użytkowników = 100.
             Przygotowany kod umieść w repozytorium GitHub. Link do repozytorium wklej poniżej i prześlij do mentora.

deseff committed:
Cześć Maciek,

nie wychodzi mi test, gdy komentarzy jest mniej niż postów. Czy mógłbyś zerknąć i coś podpowiedzieć?

Module 6.5 - Zadanie: mockujemy forum dyskusyjne
             Wyobraź sobie, że masz za zadanie dopisanie modułu prezentującego statystyki forum dyskusyjnego. Forum jeszcze nie jest ukończone, masz jednak specyfikację interfejsu, przy pomocy którego będziesz pobierać dane. Oto on (utwórz sobie pakiet com.kodilla.testing.forum.statistics w katalogu src/main/java):

             package com.kodilla.testing.forum.statistics;

             import java.util.List;

             public interface Statistics {
                 List<String> usersNames(); // list of users names
                 int postsCount();          // total quantity of forum posts
                 int commentsCount();       // total quantity of forum comments
             }
             Pamiętaj, że brak public przed nazwą metody w interfejsie również oznacza, że jest ona publiczna. Interfejs nie może posiadać innych metod, niż publiczne, więc nie trzeba tego pisać.

             W pakiecie com.kodilla.testing.forum.statistics utwórz klasę obliczającą następujące statystyki:

             liczbę użytkowników,
             liczbę postów,
             liczbę komentarzy,
             średnią liczbę postów na użytkownika,
             średnią liczbę komentarzy na użytkownika,
             średnią liczbę komentarzy na post.
             Stworzona klasa powinna posiadać metodę calculateAdvStatistics(Statistics statistics), która obliczy podane powyżej wartości i zapamięta je we właściwościach (polach) klasy. Możesz również dopisać metodę showStatistics(), która wyświetli zapamiętane we właściwościach statystyki.

             Następnie utwórz pakiet com.kodilla.testing.forum.statistics w katalogu src/test/java i stwórz zestaw testów, zawierający testy sprawdzające metodę calculateAdvStatistics(Statistics statistics) testowanej klasy. Testy zrealizuj przy pomocy mocka interfejsu Statistics. Przetestuj poprawność obliczeń wartości średnich dla różnych przypadków:

             gdy liczba postów = 0,
             gdy liczba postów = 1000,
             gdy liczba komentarzy = 0,
             gdy liczba komentarzy < liczba postów,
             gdy liczba komentarzy > liczba postów,
             gdy liczba użytkowników = 0,
             gdy liczba użytkowników = 100.
             Przygotowany kod umieść w repozytorium GitHub. Link do repozytorium wklej poniżej i prześlij do mentora.

MARudnicki commented on src/test/java/com/kodilla/testing/forum/statistics/StatisticsTestSuite.java:
```diff
@@ -0,0 +1,110 @@
+
+public class StatisticsTestSuite {
+    private Statistics statisticsMock;
+    private ForumStatistics forumStatistics;
```
te zmienne sa niewykorzystane

MARudnicki commented on src/test/java/com/kodilla/testing/forum/statistics/StatisticsTestSuite.java:
```diff
@@ -0,0 +1,110 @@
+package com.kodilla.testing.forum.statistics;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
```
nieuzywany import do wyrzucenia

MARudnicki commented on src/main/java/com/kodilla/testing/forum/statistics/ForumStatistics.java:
```diff
@@ -0,0 +1,84 @@
+
+    private double calculateAvgPostPerUser() {
+        if(usersQuantity > 0) {
+            avgPostPerUser = postsQuantity / usersQuantity;
```
dzielisz tutaj int / int, a nastepnie przypisujesz do double.
Dla Javy int / int daje int'a. Pozniejsze przypisanie jest juz wartosci zaokrąglonej do int'a
Mozesz zapisac np `avgPostPerUser = (double)postsQuantity / usersQuantity;`

MARudnicki replied:
Pamietaj o poprawie zostalych avg..

MARudnicki commented on src/test/java/com/kodilla/testing/forum/statistics/StatisticsTestSuite.java:
```diff
@@ -0,0 +1,110 @@
+        Assert.assertEquals(20, forumStatistics.getAvgCommentsPerUser(), 0.01);
+        Assert.assertEquals(2, forumStatistics.getAvgCommentsPerPost(), 0.01);
+    }
+    @Test
```
tu powinna byc linia przerwy. Uzyj reformatora kodu w intelij (Code > Reformat Code)

deseff committed:
Corrections applied:
- unused imports removed,
- unused variables removed,
- type casting (from int to double) applied.

Cześć Maciek,

nie wychodzi mi test, gdy komentarzy jest mniej niż postów. Czy mógłbyś zerknąć i coś podpowiedzieć?

Module 6.5 - Zadanie: mockujemy forum dyskusyjne
             Wyobraź sobie, że masz za zadanie dopisanie modułu prezentującego statystyki forum dyskusyjnego. Forum jeszcze nie jest ukończone, masz jednak specyfikację interfejsu, przy pomocy którego będziesz pobierać dane. Oto on (utwórz sobie pakiet com.kodilla.testing.forum.statistics w katalogu src/main/java):

             package com.kodilla.testing.forum.statistics;

             import java.util.List;

             public interface Statistics {
                 List<String> usersNames(); // list of users names
                 int postsCount();          // total quantity of forum posts
                 int commentsCount();       // total quantity of forum comments
             }
             Pamiętaj, że brak public przed nazwą metody w interfejsie również oznacza, że jest ona publiczna. Interfejs nie może posiadać innych metod, niż publiczne, więc nie trzeba tego pisać.

             W pakiecie com.kodilla.testing.forum.statistics utwórz klasę obliczającą następujące statystyki:

             liczbę użytkowników,
             liczbę postów,
             liczbę komentarzy,
             średnią liczbę postów na użytkownika,
             średnią liczbę komentarzy na użytkownika,
             średnią liczbę komentarzy na post.
             Stworzona klasa powinna posiadać metodę calculateAdvStatistics(Statistics statistics), która obliczy podane powyżej wartości i zapamięta je we właściwościach (polach) klasy. Możesz również dopisać metodę showStatistics(), która wyświetli zapamiętane we właściwościach statystyki.

             Następnie utwórz pakiet com.kodilla.testing.forum.statistics w katalogu src/test/java i stwórz zestaw testów, zawierający testy sprawdzające metodę calculateAdvStatistics(Statistics statistics) testowanej klasy. Testy zrealizuj przy pomocy mocka interfejsu Statistics. Przetestuj poprawność obliczeń wartości średnich dla różnych przypadków:

             gdy liczba postów = 0,
             gdy liczba postów = 1000,
             gdy liczba komentarzy = 0,
             gdy liczba komentarzy < liczba postów,
             gdy liczba komentarzy > liczba postów,
             gdy liczba użytkowników = 0,
             gdy liczba użytkowników = 100.
             Przygotowany kod umieść w repozytorium GitHub. Link do repozytorium wklej poniżej i prześlij do mentora.
